gbCGAZBuild.py

```diff
@@ -1,366 +1,324 @@
+            "Macedonia": "MKD",
+            "Marshall Is": "MHL",
+            "Micronesia, Fed States of": "FSM",
+            "Moldova": "MDA",
+            "Sao Tome & Principe": "STP",
+            "Solomon Is": "SLB",
+            "St Kitts & Nevis": "KNA",
+            "St Lucia": "LCA",
+            "St Vincent & the Grenadines": "VCT",
+            "Syria": "SYR",
+            "Tanzania": "TZA",
+            "Vatican City": "VAT",
+            "United States": "USA",
+            "Antarctica": "ATA",
+            "Bolivia": "BOL",
+            "Brunei": "BRN",
+            "Russia": "RUS",
+            "Trinidad & Tobago": "TTO",
+            "Swaziland": "SWZ",
+            "Venezuela": "VEN",
+            "Vietnam": "VNM",
+            "Burma": "MMR",
+        }
+        switcher_match = switcher.get(country, None)
+
+        isoCSV_match = isoCSV[isoCSV["Name"] == country]
+        if len(isoCSV_match) == 1:
+            isoCSV_match = isoCSV[isoCSV["Name"] == country]["Alpha-3code"].values[0]
+            log.debug(f"csv {isoCSV_match} {country} {isoCSV_match}")
+            return isoCSV_match
+
+        log.debug(f"swi {switcher_match} {country}")
+        return switcher_match
+
+    G["ISO_CODE"] = G.COUNTRY_NA.map(isoLookup)
+
+    # check for nulls in ISO_CODE
+    features_without_iso_code = G[G.ISO_CODE.isna() | G.ISO_CODE == ""]
+    if len(features_without_iso_code) > 0:
+        print("Error - no match.")
+        print(features_without_iso_code)
+        sys.exit(1)
+
+    disputedG.COUNTRY_NA = disputedG.COUNTRY_NA.str.replace(" (disp)", "", regex=False)
+    disputedG["ISO_CODE"] = disputedG.COUNTRY_NA.map(isoLookup)
+    disputedG[disputedG.ISO_CODE.isna() | disputedG.ISO_CODE == ""].ISO_CODE = "None"
+
+    G.to_file("./baseISO.geojson", driver="GeoJSON")
+    disputedG.to_file("./disputedISO.geojson", driver="GeoJSON")
+
+
+def process_geometries(log, args):
+    # reload with the dissolve applied
+    G = geopandas.read_file(f"{outPath}baseISO.geojson")
+
+    adm0str = ""
+    adm1str = ""
+    adm2str = ""
+    for i in G.index:
+        g = G.loc[[i]]
+        adm0str, adm1str, adm2str = process_geometry(args, g, adm0str, adm1str, adm2str)
+
+    return adm0str, adm1str, adm2str
+
+
+def process_geometry(args, g, adm0str, adm1str, adm2str):
+
+    curISO = g["ISO_CODE"].values[0]
+
+    DTA_A0Path = outPath + "ADM0_" + curISO + ".geojson"
+    g.to_file(DTA_A0Path, driver="GeoJSON")
+
+    # Get the ADM1 and ADM2 from geoBoundaries.
+    # There should be only RARE exceptions where ADM1 and ADM2 do not exist.
+    A0Path = gBPath + curISO + "/ADM0/geoBoundaries-" + curISO + "-ADM0.geojson"
+    A1Path = gBPath + curISO + "/ADM1/geoBoundaries-" + curISO + "-ADM1.geojson"
+    A2Path = gBPath + curISO + "/ADM2/geoBoundaries-" + curISO + "-ADM2.geojson"
+    # If they do not exist use the nearest layer to not have gaps in the world
+    if not os.path.isfile(A1Path):
+        A1Path = A0Path
```

Comment:
or should this be `DTA_A0Path` ?

Reply:
A0Path is correct here.  Basically, if a ADM2 or ADM1 doesn't exist, we still gap-fill using ADM0 to give the contigious coverage.